### ISSUE TITLE
Google: Error exchanging access code with OAuth2 provider

### ISSUE THREAD

**Issue by willnix86** [realm/realm-java#7119]
This is happening across Realm SDKs so just logging here to keep track. From what I gather online, there is a breaking change on Google's side that needs to be updated on the Realm side.

#### Goal
Logging into Realm via Google OAuth2

#### Actual Results
Login failed. Error message: {message: 'error exchanging access code with OAuth2 provider', code: 47}

Error message from MongoDB Realm logs.
![89660471-60950500-d89f-11ea-9e5b-7a755b03ce5e](https://user-images.githubusercontent.com/33470216/93784272-6e55de00-fbf2-11ea-8975-18e3eed9699f.png)

#### Steps & Code to Reproduce
Create new credentials in Google developer console
Enabled the auth provider on Realm.
Follow documentation
Fails here: `app.login(googleCredentials) / appp.loginAsync(googleCredentials)`

#### Version of Realm and tooling

Realm version(s): io.realm:realm-gradle-plugin:10.0.0-BETA.6

Realm Sync feature enabled: Yes

**Comment by cmelchior** [comment]
@willnix86 Thank you for reporting this. Exactly what token are you passing into `Credentials.google("...")` ?

**Comment by cmelchior** [comment]
We might have a bug in our Javadoc, can you try to send the `authCode` instead of the access token?

**Comment by willnix86** [comment]
```
    private fun getCredentials(HTTP_TRANSPORT: NetHttpTransport): Credential {
        val inStream = this::class.java.getResourceAsStream(CREDENTIALS_FILE_PATH)
            ?: throw FileNotFoundException(Lemmi.context.getString(R.string.fileNotFoundError))
        val clientSecrets =
            GoogleClientSecrets.load(JSON_FACTORY, InputStreamReader(inStream))

        val flow = GoogleAuthorizationCodeFlow.Builder(
            HTTP_TRANSPORT,
            JSON_FACTORY,
            clientSecrets,
            SCOPES
        )
            .setDataStoreFactory(FileDataStoreFactory(getTokenFolder()))
            .setAccessType("offline")
            .build()
        val ab: AuthorizationCodeInstalledApp =
            object : AuthorizationCodeInstalledApp(flow, LocalServerReceiver()) {
                @Throws(IOException::class)
                override fun onAuthorization(authorizationUrl: AuthorizationCodeRequestUrl) {
                    val url = authorizationUrl.build()
                    val browserIntent = Intent(Intent.ACTION_VIEW, Uri.parse(url))
                    startActivity(browserIntent)
                }
            }
        return ab.authorize("user").setAccessToken("user")
    }

    inner class AsyncGetCredentials: AsyncTask<String?, String, User>() {
        override fun doInBackground(vararg params: String?): User {
            val HTTP_TRANSPORT = NetHttpTransport()
            val googleCredentials =
                Credentials.google(getCredentials(HTTP_TRANSPORT).accessToken)
            return Lemmi.realmApp.login(googleCredentials)
        }

        override fun onPostExecute(result: User?) {
            super.onPostExecute(result)
            myHandler = Handler()
            myHandler.postDelayed({
                goToMainActivity()
            },splashTime)
        }
    }
```

This is the code snippet - not sure how to send the authCode instead (not sure it's a property I get back)

**Comment by cmelchior** [comment]
After looking into this. Google apparently changed some things with regard to how we exchange the authentication code to a property access token. We are currently exploring options with them on how to fix this.

**Comment by tsuharesu** [comment]
Was this fixed? I think I'm still having this issue with `io.realm:realm-gradle-plugin:10.2.0` on Android.

**Comment by rorbech** [comment]
@tsuharesu This was indeed fixed in 10.2.0, but you have to explicitly indicate that you are using an `ID_TOKEN` with 
https://docs.mongodb.com/realm-sdks/java/10.2.0/io/realm/mongodb/Credentials.html#google-java.lang.String-io.realm.mongodb.auth.GoogleAuthType-
```
val creds = Credentials.google("google-token", GoogleAuthType.ID_TOKEN)
```
Otherwise it will default to interpret the argument as an `AUTH_CODE`. 

Let us know if you still have issues.
